src/VisualStudio/Core/Def/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -138,30 +143,31 @@ public void ClearErrors(ProjectId projectId)
 
         // Update the state to clear diagnostics and raise corresponding diagnostic updated events
         // on a serialized task queue.
-        _taskQueue.ScheduleTask(nameof(ClearErrors), () =>
+        _taskQueue.AddWork(async cancellationToken =>
         {
             if (state == null)
             {
                 // TODO: Is it possible that ClearErrors can be invoked while the build is not in progress?
                 // We fallback to current solution in the workspace and clear errors for the project.
-                ClearErrorsCore(projectId, _workspace.CurrentSolution, state);
+                await ClearErrorsCoreAsync(projectId, _workspace.CurrentSolution, state, cancellationToken).ConfigureAwait(false);
             }
             else
             {
+                if (state.CancellationToken.IsCancellationRequested is true)
+                    return;
```

Comment:
this logic is intentionally added. in the original code, we would 'ScheduleTask' using `GetApplicableCancellationToken(state)` below.  That meant that if this partiular 'state' instance was canceled, we would not perform this work.  Checking its cancelation token here accomplishes teh same.